### ISSUE TITLE
Better Building: A List Of 2 Requests

### ISSUE THREAD

**Issue by jonnyawsom3** [cabaletta/baritone#985]
## Describe your suggestion
1.More useful errors when the build command results in "unable to do it", for example, saying if there is not enough height available for the schematic (an issue that i think I've been having multiple times, when attempting to make the [jesus statue](https://youtu.be/CgS-spZPKxc), but i have no way of knowing)

2.A simple form of size preview, possibly in the form of a bounding box to show the size of the schematic

## Settings
bounding box options such as: 
3 axis (a line going along each axis)
3 axis wall (a single "wall" for x y and z)
full (bounding box covering the size of the entire schematic)
changed via a "boundingbox" command

## Context
To assist in placement of schematics using the build command

## Final checklist
- [ ] I have not used any OwO's or UwU's in this issue.

**Comment by Hexxonite** [comment]
Agree on that first point.

However imo it would be enough to have the "size preview" in chat like the starting position.
You will have to do a little math for correct positioning anyway. And this way it wont easily break rendering...

Also I'd add a 3rd request: Building block list.
This could be integrated to show which blocks and how many are missing when its "unable to do it" aswell.

Maybe looking something like this:
![image](https://user-images.githubusercontent.com/5312542/65680747-e0cfc480-e057-11e9-92e2-f9e865123e64.png)

**Comment by jonnyawsom3** [comment]
I definitely agree with the material list as i was going to make that number 3 on my list. Although I thought my bounding box idea was already asking a lot, so i had left it out
A size preview in chat doesn't make much sense to me since then you'd still have to go measure out what it would have to break to build it (assuming you're not in a flat area), _but_ i guess anything is better than nothing

**Comment by jonnyawsom3** [comment]
i just realised that this is pretty much the whole point of schematica and it's integration...

**Comment by enescaglayan** [comment]
i also agree this, one time i loaded a schematic and it didnt show me what blocks are required to do that. Also it should not only use blocks in hotbar also use things in my Inventory.

**Comment by Hexxonite** [comment]
> Also it should not only use blocks in hotbar also use things in my Inventory.

you can always enable "# allowInventory"

**Comment by jonnyawsom3** [comment]
Edited/updated my suggestion to be more reasonable

**Comment by enescaglayan** [comment]
Good

2019-10-12 16:52 GMT+03:00, jonnyawsom3 <notifications@github.com>:
